pt/licoes/introducao-jupyter-notebooks.md

```diff
@@ -0,0 +1,427 @@
+ 
+ 
+```
+with open('ph-jupyter-notebook-example.csv') as f:
+  csv_reader = csv.reader(f, delimiter=',')
+    for row in csv_reader:
+        parseddate = dateutil.parser.parse(row[1])
+            print(parseddate)
+
+ ```
+
+Execute a célula com o código que você acabou de adicionar. Pode levar mais tempo; continue esperando até que o asterisco ao lado da célula de código se transforme em um número. O resultado deve mostrar a lista de datas de publicação, formatadas de forma diferente, com hífen em vez de barras, e com a adição das horas, minutos e segundos (como zeros, porque as datas registradas não incluem esses dados). À primeira vista, parece que funcionou, mas se você compará-lo mais de perto com o ficheiro de origem, você verá que o módulo dateutil não está sendo consistente em como analisa as datas. Datas em que o valor do dia é maior que 12 estão sendo analisadas corretamente (ele sabe que um valor maior que 12 não pode ser um mês), mas quando o valor da data é 12 ou menos, a data está sendo identificada com o mês primeiro. A primeira linha do ficheiro de origem tem a data 1/7/18, que é entendida como "2018-01-07 00:00:00". Na documentação para dateutil, você descobrirá que você pode [especificar `dayfirst=true`](https://perma.cc/W54E-SP5Z) para corrigir isso. Edite a última célula de código e altere a penúltima linha para ler:
+
+```
+parseddate = dateutil.parser.parse(row[1], dayfirst=True)
+
+ ```
+
+Quando você executar a linha novamente, você verá que todas as datas foram analisadas corretamente. 
+
+Analisar a data é apenas o primeiro passo – você ainda precisa usar o módulo datetime para converter as datas em dias da semana.
+
+Exclua a última linha do bloco de código e substitua-a pelo seguinte (certificando-se de que você tenha o mesmo nível de recuo da última linha anterior, para ambas as linhas):
+
+  
+```
+dayofweek = datetime.date.strftime(parseddate, '%A')
+
+print(dayofweek)
+
+```
+  
+Execute o bloco de códigos novamente. Isso deve lhe dar uma lista de dias da semana.
+
+Agora que você tem código para analisar e re-formatar uma data, você precisa fazê-lo para ambas as datas em cada linha do seu ficheiro de origem. Porque você sabe que tem código funcionante na célula de código atual, se você não se sentir muito confortável com Python, você pode querer copiar a célula de código atual antes de fazer modificações. Selecione a célula que deseja copiar e clique no botão copiar na barra de ferramentas; o botão de colar irá colar a célula abaixo de qualquer célula atualmente selecionada. Fazer uma cópia permite que você faça livremente alterações no código, sabendo que você sempre pode voltar facilmente para uma versão que funciona.
+
+Se você não quiser resolver isso por conta própria, você pode copiar e colar esse código em uma nova célula de código ou substituir a célula de código atual:
+
+```

```

Comment:
Corrigir indetação do bloco para:

```
#identifies the source file to open, calls it f
with open('ph-jupyter-notebook-example.csv') as f:
  #creates an output file (referred to as "out" in the notebook) for you to write to
 with open('ph-jupyter-notebook-example-dayofweek.csv', 'w') as out:
    #defines "csv_reader" as running the function csv.reader on the file
    csv_reader = csv.reader(f, delimiter=',')
    #defines "csv_writer" as running the functin csv.writer to "out" (the output file)
    csv_writer = csv.writer(out)
    #for each row that's being read by csv_reader...
    for row in csv_reader:
        #defines "csv_reader" as running the function csv.reader on the file
        csv_reader = csv.reader(f, delimiter=',')
        #for each row that's being read by csv_reader...
        for row in csv_reader:
            #creates a list called "values" with the contents of the row
            values = list(row)
            #defines "rating" as the first thing in the list
            #counting in Python starts with 0, not 1
            rating = values[0]
            #defines "parseddatepub" as the second thing (1, because we start with 0) in the list,
            #converted into a standard date format using dateutil.parser
            #and when those dates are parsed, the parser should know
            #that the first value in the sequence is the day
            parseddatepub = dateutil.parser.parse(values[1], dayfirst=True)
            #same as above for the updated date, the third thing (2) in the list
            parseddateupdate = dateutil.parser.parse(values[2], dayfirst=True)
            #defines "dayofweekpub" as parseddatepub (defined above), converted to the day of week
            #%A is what you use to change it to the day of the week
            #You can see othe formats here: https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior
            dayofweekpub = datetime.date.strftime(parseddatepub, '%A')
            #same thing for update date
            dayofweekupdate = datetime.date.strftime(parseddateupdate, '%A')
            #creates a list of the rating and the newly formatted dates
            updatedvalues = [rating, dayofweekpub, dayofweekupdate]
            #writes all the values under this code cell
            csv_writer.writerow(updatedvalues)
            print(updatedvalues)
```

Reply:
Os comentários do código também não devem ser traduzidos?